### PR TITLE
IGAPP-829: Upgrade dependencies and fix security vulnerabilities

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

**Comment by klinzo** [comment]
Not all security vulnerabilities fixable right now. Updated all dependencies except:
* react-navigation (https://issues.tuerantuer.org/browse/IGAPP-834)
* react-router (https://issues.tuerantuer.org/browse/IGAPP-833)
* jest dependencies
* react-native
* react-native-webview

File: web/src/routes/SearchPage.tsx [hunk 1]
```diff
@@ -112,6 +109,8 @@ const SearchPage = ({ match, cityModel, location, languages, history }: PropsTyp
       category1.model.title.localeCompare(category2.model.title)
     )
 
+  parser.end()
```
Comment (klinzo) [inline web/src/routes/SearchPage.tsx:112]: Bug prevented by our unit tests @maxammann 🎉

**Comment by klinzo** [comment]
Please test carefully on iOS, android and web, did not encounter any problems though.
Notable things done in this PR:
* Many dependency upgrades
* Replaced rn-fetch-blob with react-native-blob-util (which is a maintained fork and usable without changes) [changes](https://github.com/Integreat/integreat-app/pull/499/commits/d39c8d01d5d25a2db44f5f119fd34443bd2be352)
* Prettier upgrade (leads to new rules/formatting) [changes](https://github.com/Integreat/integreat-app/pull/499/files/6ddd511ad1919b24e2814de4bbf9757ff25abcbb..6d702fca537f815020bbdc72e65c3b570f22dff1)

**Review by SteffiStuffel — COMMENTED** [review]
Tested this on web. Everything fine.

Tested this on Android Emulator, got this error, but everything it was working nevertheless
i18error

Also please adjust the readme to the now needed node version:
node-version

**Review by Taggotty — APPROVED** [review]

**Comment by Taggotty** [comment]
and tested on chrom and android (same problem as steffi)

**Review by f1sh1918 — COMMENTED** [review]
Works on android emulator (same i18n error as mentioned), works on web (Chrome & FF)
Works on ios emulator but on ios native i get a metro bundler error which i also faced on android before deleting node modules and cleaned cache. On IOS native it still exists
```js
Failed to construct transformer:  Error: error:0308010C:digital envelope routines::unsupported
    at new Hash (node:internal/crypto/hash:67:19)
    at Object.createHash (node:crypto:130:10)
    at stableHash (/Users/afischer/projects/integreat-app/native/node_modules/metro-cache/src/stableHash.js:19:8)
    at Object.getCacheKey (/Users/afischer/projects/integreat-app/native/node_modules/metro-transform-worker/src/index.js:593:7)
    at getTransformCacheKey (/Users/afischer/projects/integreat-app/native/node_modules/metro/src/DeltaBundler/getTransformCacheKey.js:24:19)
    at new Transformer (/Users/afischer/projects/integreat-app/native/node_modules/metro/src/DeltaBundler/Transformer.js:48:9)
    at /Users/afischer/projects/integreat-app/native/node_modules/metro/src/Bundler.js:22:29
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  opensslErrorStack: [ 'error:03000086:digital envelope routines::initialization error' ],
  library: 'digital envelope routines',
  reason: 'unsupported',
  code: 'ERR_OSSL_EVP_UNSUPPORTED'
}
```

File: native/ios/Integreat.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -379,6 +379,9 @@
 			buildActionMask = 2147483647;
 			files = (
 			);
+			inputPaths = (
+				"$(SRCROOT)/$(BUILT_PRODUCTS_DIR)/$(INFOPLIST_PATH)",
+			);
```
[the remark this answers is not in the document]
Comment (klinzo) [reply native/ios/Integreat.xcodeproj/project.pbxproj:384]: Not sure, was automatically added by xcode

**Comment by klinzo** [comment]
> Works on android emulator (same i18n error as mentioned), works on web (Chrome & FF) Works on ios emulator but on ios native i get a metro bundler error which i also faced on android before deleting node modules and cleaned cache. On IOS native it still exists. Maybe it depends on my env or did anyone ever face this?

I have never seen this tbh. Did you also delete your pods?

**Comment by f1sh1918** [comment]
> > Works on android emulator (same i18n error as mentioned), works on web (Chrome & FF) Works on ios emulator but on ios native i get a metro bundler error which i also faced on android before deleting node modules and cleaned cache. On IOS native it still exists. Maybe it depends on my env or did anyone ever face this?
> 
> I have never seen this tbh. Did you also delete your pods? On my MacBook (when running it on simulator) and in CI it seems to work just fine.

looks like a node issues. Gonna check it now!

**Comment by klinzo** [comment]
> Tested this on web. Everything fine.
> 
> Tested this on Android Emulator, got this error, but everything it was working nevertheless i18error
> 
> Also please adjust the readme to the now needed node version: node-version

Downgraded i18next, will need to be done when switching from moment to intl.

**Comment by klinzo** [comment]
@SteffiStuffel could you check that the error does not happen anymore pls?

**Comment by SteffiStuffel** [comment]
> @SteffiStuffel could you check that the error does not happen anymore pls?

Yes, it is fixed now.

**Review by f1sh1918 — APPROVED** [review]
Great work! tested on android emulator without any issues, also works on ios native

**Review by SteffiStuffel — APPROVED** [review]

**Pull request merged**